### PR TITLE
Add test cases to verify apache/dubbo#13751

### PR DESCRIPTION
**Pull request by aofall**
verify apache/dubbo#13751

### CODE REVIEW

**Comment by aofall** [comment]
@AlbumenJ PTAL

File: 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/main/java/org/dubbo/samples/test/consumer/CustomFilter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
+        return invoker.invoke(invocation).whenCompleteWithContext(
+                (r, t) -> {
+                    RpcServiceContext serviceContext = RpcContext.getServiceContext();
+                    String localAddress = serviceContext.getLocalAddressString();
+                    String remoteAddress = serviceContext.getRemoteAddressString();
+                    System.out.println("自定义日志local:" + localAddress);
+                    System.out.println("自定义日志remote:" + remoteAddress);
```
Comment (AlbumenJ) [inline 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/main/java/org/dubbo/samples/test/consumer/CustomFilter.java:43]: `System.out.println` is useless in test

File: 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/test/java/org/apache/dubbo/samples/test/ConsumerIT.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                .application(new ApplicationConfig("consumer"))
+                .reference(reference1)
+                .reference(reference2)
+                .consumer(consumerConfig)
+                .start();
+
+        DemoService demoService = reference1.get();
+        demoService.sayHello(new User("dubbo"));
+
+        int dubboExpected = CustomFilter.expected();
```
Comment (AlbumenJ) [inline 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/test/java/org/apache/dubbo/samples/test/ConsumerIT.java:54]: Also test address, not only port

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/main/java/org/dubbo/samples/test/consumer/CustomFilter.java (hunk 1 above)
Comment (aofall) [reply 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/main/java/org/dubbo/samples/test/consumer/CustomFilter.java]: Done

File: 99-integration/dubbo-samples-test-13751/pom.xml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    <name>Dubbo Samples Test for 13751</name>
+    <description>Dubbo Samples Test for 13751</description>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <dubbo.version>3.2.12-SNAPSHOT</dubbo.version>
```
Comment (AlbumenJ) [inline 99-integration/dubbo-samples-test-13751/pom.xml:41]: Depend on the release version of Dubbo here. The test suite will replace it dynamically when testing.

File: 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/test/java/org/apache/dubbo/samples/test/ConsumerIT.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                .reference(reference2)
+                .consumer(consumerConfig)
+                .start();
+
+        CustomFilter.resetExpected();
+        DemoService demoService = reference1.get();
+        demoService.sayHello(new User("dubbo"));
+
+        String dubboExpectHost = CustomFilter.expectedHost();
+        Assert.assertNotNull(dubboExpectHost);
```
Comment (AlbumenJ) [inline 99-integration/dubbo-samples-test-13751/dubbo-samples-test-13751-consumer/src/test/java/org/apache/dubbo/samples/test/ConsumerIT.java:56]: Check if not localhost

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: 99-integration/dubbo-samples-test-13751/pom.xml (hunk 3 above)
Comment (aofall) [reply 99-integration/dubbo-samples-test-13751/pom.xml:41]: Change it into 3.2.12 or others?

Comment (AlbumenJ) [reply 99-integration/dubbo-samples-test-13751/pom.xml:41]: > Change it into 3.2.12 or others?

3.2.12

Comment (aofall) [reply 99-integration/dubbo-samples-test-13751/pom.xml]: Done

**Pull request merged**